Queries/Transactions/Handlers/GetTransactionsHandler.cs

```diff
@@ -0,0 +1,102 @@
+        private readonly IMapper _mapper;
+
+        public GetTransactionsHandler(IDbContextFactory<HaSpManContext> contextFactory, IMapper mapper)
+        {
+            _contextFactory = contextFactory;
+            _mapper = mapper;
+        }
+        public async Task<Paginated<TransactionSummary>> Handle(GetTransactionQuery request, CancellationToken cancellationToken)
+        {
+            var context = _contextFactory.CreateDbContext();
+            var transactions = context.Transactions.AsNoTracking()
+                .Where(GetFilterCriteria(request.SearchString));
+
+            var totalCount = await transactions.CountAsync(cancellationToken);
+
+            transactions = GetOrderedQueryable(request, transactions);
+
+            var transactionViewModels = 
+                transactions
+                    .ProjectTo<TransactionSummary>(_mapper.ConfigurationProvider)
+                    .Skip(request.PageIndex * request.PageSize)
+                    .Take(request.PageSize);
+
+            var items = await transactionViewModels.ToListAsync(cancellationToken);
+
+            return new Paginated<TransactionSummary>()
+            {
+                Items = items,
+                Total = totalCount
+            };
+
+        }
+
+        private IQueryable<Transaction> GetOrderedQueryable(GetTransactionQuery request, IQueryable<Transaction> transactions)
+        {
+            transactions = request.SortDirection switch
+            {
+                SortDirection.Ascending => request.OrderBy switch
+                {
+                    TransactionSummaryOrderOption.From => transactions.OrderBy(x => x.CounterPartyName),
+                    TransactionSummaryOrderOption.Amount => transactions.OrderBy(x => x.Amount),
+                    TransactionSummaryOrderOption.ReceivedDateTime => transactions.OrderBy(x => x.ReceivedDateTime),
+                    _ => transactions.OrderBy(x => x.DateFiled)
+                },
+                SortDirection.Descending => request.OrderBy switch
+                {
+                    TransactionSummaryOrderOption.From => transactions.OrderByDescending(x => x.CounterPartyName),
+                    TransactionSummaryOrderOption.Amount => transactions.OrderByDescending(x => x.Amount),
+                    TransactionSummaryOrderOption.ReceivedDateTime => transactions.OrderByDescending(x =>
+                        x.ReceivedDateTime),
+                    _ => transactions.OrderByDescending(x => x.DateFiled)
+                },
+                _ => transactions
+            };
+
+            return transactions;
+        }
+
+        public Expression<Func<Transaction, bool>> GetFilterCriteria(string searchString)
+        {
+            return t => t.CounterPartyName.ToLower().Contains(searchString) ||
+                        t.Amount.ToString().Contains(searchString);
+        } 
+    }
+
+    public record GetTransactionQuery(string SearchString, int PageIndex, int PageSize,
```

Comment:
All other queries are defined in their own files and namespace, I would like to keep this consistent